src/commands/add.rs

```diff
@@ -105,8 +138,41 @@ pub async fn add(args: Add, global: GlobalArgs) -> Result<()> {
                 "The 'dependencies' section must be an array",
             )
         })?;
+    let mut added_deps = Vec::new();
     for dep in deps.iter() {
-        insert_formatted(dependencies, dep.to_string());
+        let mut merge_list = remove_matching_dependencies(dependencies, &dep.name)?;
+        let to_insert = if merge_list.is_empty() {
+            added_deps.push(dep.clone());
+            dep.clone()
+        } else if merge_list.len() == 1 && !requirement_needs_update(&merge_list[0], dep) {
+            merge_list[0].clone()
+        } else {
+            merge_list.push(dep.clone());
+            let merged = merge_list
+                .iter()
+                .skip(1)
+                .try_fold(merge_list[0].clone(), |merged, next| {
```

Comment:
Isn't this a nightly feature?

Reply:
not on the latest stable https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.try_fold

Reply:
Must have looked at an older documentation page earlier today then 😅